tests/x11regressions/gnomecase/gnome_default_applications.pm

```diff
@@ -19,55 +19,82 @@ sub run {
     # Prepare test files
     x11_start_program("xterm");
     assert_screen 'xterm';
+    if (
+        check_default_apps(
+            # mimetype, application (to confirm if opens in the right application)
+            ['image/jpg',           'eog.desktop'],
+            ['image/png',           'eog.desktop'],
+            ['application/pdf',     'evince.desktop'],
+            ['application/x-bzip2', 'org.gnome.FileRoller.desktop'],
+            ['application/gzip',    'org.gnome.FileRoller.desktop']))
+    {
+        assert_script_run "mkdir gnometest";
+        assert_script_run "wget -P /home/$username/gnometest " . autoinst_url . "/data/x11regressions/test.pdf";
+        assert_script_run "wget -P /home/$username/gnometest " . autoinst_url . "/data/x11regressions/shotwell_test.jpg";
+        assert_script_run "wget -P /home/$username/gnometest " . autoinst_url . "/data/x11regressions/shotwell_test.png";
+        assert_script_run "cp /usr/share/w3m/w3mhelp.html /home/$username/gnometest/";
+        assert_script_run "tar cjvf /home/$username/gnometest/test.tar.bz2 -C /home/$username/gnometest/ test.pdf";
+        assert_script_run "tar czvf /home/$username/gnometest/test.tar.gz -C /home/$username/gnometest/ test.pdf";
+        send_key 'alt-f4';
```

Comment:
to avoid indenting everything I suggest to put the old code into a function called from within here